### PR TITLE
doc(types): Remove independent slices pattern

### LINKED ISSUE #1046: Suggestion from my own confusion
The docs say to use  this `StateCreator<MySlice, [["zustand/devtools", never]], []>` for independent slices when using middleware but the docs don't show how to use it in an example. Couple that with trying to partialize certain things and I believe there should be clarity there. 

If I have two slices (independent) and want to persist certain things from each one of the two slices, it may be helpful to show the intended way to accomplish this. 

Thanks a lot for your time!

### PR DESCRIPTION
**Pull request by devanshj**
Closes #1046

### CODE REVIEW

**Comment by dai-shi** [comment]
Should we somewhere have interdependent slices pattern you showed in https://github.com/pmndrs/zustand/issues/1046#issuecomment-1185977777?

**Comment by devanshj** [comment]
> Should we somewhere have interdependent slices pattern you showed in [#1046 (comment)](https://github.com/pmndrs/zustand/issues/1046#issuecomment-1185977777)?

Yes it's already there in the doc, I've only removed the independent version not interdependent. (And the one I showed in the issue is same as the one in the doc).

File: docs/typescript.md [hunk 1]
```diff
@@ -345,40 +345,7 @@ const useStore = create<
...
-  ...createFishSlice(...a),
-}))
-```
-
-If you have some middlewares then replace `StateCreator<MySlice, [], []>` with `StateCreator<MySlice, Mutators, []>`. Eg if you're using `devtools` then it'll be `StateCreator<MySlice, [["zustand/devtools", never]], []>`. See the ["Middlewares and their mutators reference"](#middlewares-and-their-mutators-reference) section for a list of all mutators.
-
-Also you can even write `StateCreator<MySlice>` instead of `StateCreator<MySlice, [], []>` as the second and third parameter have `[]` as their default value.
-
-### Interdependent slices pattern
+### Slices pattern
```
Comment (dai-shi) [inline docs/typescript.md:348]: Ah, I see. Got it now.

**Review by dai-shi — APPROVED** [review]

**Comment by rothsandro** [comment]
The link to the #independent-slices-pattern chapter that is now broken:
https://github.com/pmndrs/zustand/wiki/Splitting-the-store-into-separate-slices

Should this link to #slices-pattern instead?

**Comment by devanshj** [comment]
> Should these links refer to #slices-pattern instead?

Yes

**Comment by dai-shi** [comment]
Feel free to update the wiki.

**Pull request merged**